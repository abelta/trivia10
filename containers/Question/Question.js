import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-native';
import { Button, Heading, Row, Text, Title, View } from '@shoutem/ui';
import he from 'he';
import { right as answeredRight, wrong as answeredWrong } from '../../actions/answers';
import * as constants from '../../constants';
import { questionByRound as selectQuestionByRound } from '../../selectors/questions';
import { question as questionShape } from '../../propTypeShapes/questions';

class Question extends PureComponent {
  constructor(props) {
    super(props);
    this.answer = this.answer.bind(this);
  }

  answer(choice) {
    const { answerRight, answerWrong, question: { correctAnswer } } = this.props;
    choice === correctAnswer
      ? answerRight()
      : answerWrong();
  }

  render() {
    const { question: { category, text }, round } = this.props;
    if (round === constants.GAME_ROUNDS_TOTAL) return <Redirect to="/results" />;
    return (
      <View styleName="flexible">
        <View styleName="flexible">
          <Heading>{category}</Heading>
          <Text styleName="small">{round + 1} of {constants.GAME_ROUNDS_TOTAL}</Text>
        </View>
        <View styleName="flexible">
          <Text>{he.unescape(text)}</Text>
        </View>
        <View styleName="flexible">
          <Row>
            <Button onPress={() => this.answer(true)}><Title>TRUE</Title></Button>
            <Button onPress={() => this.answer(false)}><Title>FALSE</Title></Button>
          </Row>
        </View>
      </View>
    );
  }
}

Question.propTypes = {
  answerRight: PropTypes.func.isRequired,
  answerWrong: PropTypes.func.isRequired,
  question: questionShape.isRequired,
  round: PropTypes.number.isRequired,
};

const mapStateToProps = state => ({
  question: selectQuestionByRound(state),
  round: state.game.round,
});

const mapDispatchToProps = dispatch => ({
  answerRight: () => dispatch(answeredRight()),
  answerWrong: () => dispatch(answeredWrong()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Question);
