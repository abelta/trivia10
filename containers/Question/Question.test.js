import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Question from '.';

const state = {
  game: {
    score: 0,
    round: 0,
  },
  questions: [
    {
      category: 'History',
      difficulty: 'Hard',
      text: 'Japan was part of the Allied Powers during World War I.',
      correctAnswer: 'true',
    },
  ],
};

const store = configureStore()(state);

describe('Question', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Question />
      </Router>
    </Provider>,
  ).toJSON();

  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
