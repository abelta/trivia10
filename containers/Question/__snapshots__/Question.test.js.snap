// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question renders correctly 1`] = `
<View
  animationOptions={Object {}}
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
  styleName="flexible"
  virtual={undefined}
>
  <View
    animationOptions={Object {}}
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    styleName="flexible"
    virtual={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      animationOptions={Object {}}
      disabled={false}
      ellipsizeMode="tail"
      onLayout={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#222222",
          "fontFamily": "Rubik-Regular",
          "fontSize": 25,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": 30,
        }
      }
      virtual={undefined}
    >
      History
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      animationOptions={Object {}}
      disabled={false}
      ellipsizeMode="tail"
      onLayout={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#666666",
          "fontFamily": "Rubik-Regular",
          "fontSize": 15,
          "fontStyle": "normal",
          "fontWeight": "normal",
        }
      }
      styleName="small"
      virtual={undefined}
    >
      1
       of 
      10
    </Text>
  </View>
  <View
    animationOptions={Object {}}
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    styleName="flexible"
    virtual={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      animationOptions={Object {}}
      disabled={false}
      ellipsizeMode="tail"
      onLayout={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#666666",
          "fontFamily": "Rubik-Regular",
          "fontSize": 15,
          "fontStyle": "normal",
          "fontWeight": "normal",
        }
      }
      virtual={undefined}
    >
      Japan was part of the Allied Powers during World War I.
    </Text>
  </View>
  <View
    animationOptions={Object {}}
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    styleName="flexible"
    virtual={undefined}
  >
    <View
      animationOptions={Object {}}
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 0,
          "justifyContent": "flex-start",
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        }
      }
      virtual={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#ffffff",
            "borderRadius": 2,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "paddingLeft": 15,
            "paddingRight": 5,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          animationOptions={Object {}}
          disabled={false}
          ellipsizeMode="tail"
          onLayout={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#222222",
              "fontFamily": "Rubik-Regular",
              "fontSize": 20,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": 25,
            }
          }
          virtual={undefined}
        >
          TRUE
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#ffffff",
            "borderRadius": 2,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "paddingLeft": 15,
            "paddingRight": 5,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          animationOptions={Object {}}
          disabled={false}
          ellipsizeMode="tail"
          onLayout={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#222222",
              "fontFamily": "Rubik-Regular",
              "fontSize": 20,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": 25,
            }
          }
          virtual={undefined}
        >
          FALSE
        </Text>
      </View>
    </View>
  </View>
</View>
`;
